File: backend/src/main/java/unam/ciencias/ids/playbit/models/Round.java
package unam.ciencias.ids.playbit.models;

import java.util.List;

import org.hibernate.annotations.ManyToAny;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter 
@Entity
@Table(name = "round")
public class Round {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "stage_id")
    private Stage stage;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "group_id")
    private Group group;

    @Column(name = "number")
    private int number;

    @OneToMany(fetch = FetchType.EAGER,mappedBy = "round")
    private List<Match> matches;
}

File: frontend/src/tranformers/Round.ts
import { Round } from "brackets-model"
import MyRound from "../types/Round"

class RoundT {
    to (value:  MyRound): Round {
        return {
            id: value.id != undefined ? value.id : 0,
            stage_id: value.stage.id,
            group_id: value.group.id,
            number: value.number
        }
    }

    from (value: Round): MyRound {
        return {
            stage: {
                id: value.stage_id as number
            },
            group: {
                id: value.group_id as number
            },
            number: value.number
        }
    }
}

export default RoundT